test/java/controllers/KairosDbProxyControllerTest.java

```diff
@@ -220,23 +217,90 @@ public void testAddMergeAggregator() {
+                                new Aggregator.Builder().setName("filter").build())).build())).build())
+        ).thenReturn(
                 CompletableFuture.completedFuture(new MetricsQueryResponse.Builder()
                         .setQueries(ImmutableList.of()).build())
         );
 
-        // ***
-        // Test case where one metric doesn't have merge aggregator and is not one of "min,max,avg,count,sum"
-        // ***
         final Result result = Helpers.invokeWithContext(request, Helpers.contextComponents(), () -> {
             final CompletionStage<Result> completionStage = _controller.queryMetrics();
             return completionStage.toCompletableFuture().get(10, TimeUnit.SECONDS);
         });
 
         assertEquals(Http.Status.OK, result.status());
         assertEquals("{\"queries\":[]}", Helpers.contentAsString(result));
+    }
+
+    @Test
+    public void testAddMergeAggregatorOff() {
+        when(_mockConfig.getBoolean(eq("kairosdb.proxy.addMergeAggregator"))).thenReturn(false);
+
+        final KairosDbProxyController controller = new KairosDbProxyController(
+                _mockConfig,
+                _mockWSClient,
+                _mockKairosDbClient,
+                OBJECT_MAPPER,
+                _mockMetricsFactory,
+                _mockMetricsqueryConfig
+        );
+
+        final Metric metric1 = new Metric.Builder()
+                .setName("metric1")
+                .setAggregators(ImmutableList.of(new Aggregator.Builder().setName("count").build()))
+                .build();
+        final Metric metric2 = new Metric.Builder()
+                .setName("metric2")
+                .setAggregators(ImmutableList.of(new Aggregator.Builder().setName("filter").build()))
+                .build();
+        final MetricsQuery metricsQuery = new MetricsQuery.Builder()
+                .setStartTime(Instant.now())
+                .setMetrics(ImmutableList.of(metric1, metric2))
+                .build();
+
+        final Http.RequestBuilder request = Helpers.fakeRequest()
+                .method(Helpers.POST)
+                .uri("/api/v1/datapoints/query")
+                .header("Content-Type", "application/json")
+                .bodyJson(OBJECT_MAPPER.<JsonNode>valueToTree(metricsQuery));
+
+        when(_mockKairosDbClient.queryMetrics(metricsQuery)).thenReturn(
+                CompletableFuture.completedFuture(new MetricsQueryResponse.Builder()
+                        .setQueries(ImmutableList.of()).build())
+        );
+
+        final Result result = Helpers.invokeWithContext(request, Helpers.contextComponents(), () -> {
+            final CompletionStage<Result> completionStage = controller.queryMetrics();
+            return completionStage.toCompletableFuture().get(10, TimeUnit.SECONDS);
+        });
+
+        assertEquals(Http.Status.OK, result.status());
+        assertEquals("{\"queries\":[]}", Helpers.contentAsString(result));
```

Comment:
One last ask, sorry: we now have these 18 lines repeated almost exactly across three different test-methods, differing only in: 
- what argument we're passing to `when(_mockKairosDbClient.queryMetrics(...))`
- what `controller` is being hit by the request

Could we refactor this out into a method? (This might be a great chance to experiment with IntelliJ's "Extract method" functionality! Select a bunch of code, right-click, "Refactor", "Extract method". It makes it really easy to refactor a chunk of code out into its own method.)